[core] Update maven-enforcer-plugin to require Java 11

**Is your feature request related to a problem? Please describe.**
The maven build currently require java 10, which is not supported anymore by Oracle, and not widely available. Builds on JDK 10 may not be reliable (specifically, the javadoc plugin could fail), while they work on JDK 11+. Our CI doesn't even use Java 10, but an OpenJDK 11 distribution 

**Describe the solution you'd like**
Require Java 11+. This is the latest LTS

https://github.com/pmd/pmd/blob/a12d13a1ac719ef782859c350fc639f9bc68cebe/pom.xml#L444-L446

**Describe alternatives you've considered**
None, requiring Java 10 doesn't make sense anymore

**Additional context**
Javadoc plugin failure on jdk 10:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.2.0:jar (attach-javadocs) on project pmd-cpp: MavenReportException: Error while generating Javadoc:
[ERROR] Exit code: 1 - javadoc: warning - You have not specified the version of HTML to use.
[ERROR] The default is currently HTML 4.01, but this will change to HTML5
[ERROR] in a future release. To suppress this warning, please specify the
[ERROR] version of HTML used in your documentation comments and to be
[ERROR] generated by this doclet, using the -html4 or -html5 options.
[ERROR] javadoc: error - An internal exception has occurred.
[ERROR] (com.sun.tools.javac.code.Symbol$CompletionFailure: class file for org.apache.tools.ant.Task not found)
```

--- a/pom.xml
+++ b/pom.xml
@@ -442,7 +442,7 @@
                         <configuration>
                             <rules>
                                 <requireJavaVersion>
-                                    <version>[10,)</version>
+                                    <version>[11,)</version>
                                 </requireJavaVersion>
                             </rules>
                         </configuration>

